aws-cpp-sdk-core/include/aws/core/utils/memory/stl/AWSMap.h

```diff
@@ -13,6 +13,21 @@
 #include <unordered_map>
 #include <cstring>
 
+#ifdef USE_AWS_MEMORY_MANAGEMENT
+#include <aws/core/utils/HashingUtils.h>
+/* Inject hash method for an Aws::String with a custom allocator to workaround original C++ defect:
+ * "hash support for std::basic_string with customized allocators was not enabled"
+ * see LWG 3705: https://en.cppreference.com/w/cpp/string/basic_string/hash */
+namespace std
+{
+    template<>
+    struct std::hash<Aws::String>
+    {
+        std::size_t operator()(const Aws::String& t) const { return Aws::Utils::HashingUtils::HashString(t.c_str()); }
```

Comment:
Also worth noting Aws::Crt::HashString exists in aws/crt/StringUtils.h, perhaps a simplification can be made here.  Placing this in HashingUtils.h will forward:
```
            /**
             * Forward HashString to CRT library.
             */
            inline static size_t HashString(const char* strToHash) noexcept { return Aws::Crt::HashString(strToHash); }
```

Reply:
I'm wondering why Aws::Crt::HashString does not use a better hashing algorithm like [SipHash](https://github.com/veorq/SipHash).